File: RockPaperScissors/CPP/src/RockPaperScissors.cpp
/*******************************************************************
* Name: Michael Nutt(michaeltnutt@gmail.com)
* Desc: Rock Paper Scissors game, takes input from one human player,
*       validates correct input and then generates a random number
*       for the computer to choose a move. Keeps score of wins
*******************************************************************/

#include <iostream>
#include <string>
#include "Human.h"
#include "CPU.h"
#include "Score.h"
#include "Game.h"

int main()
{
  Human *p1 = new Human();
  CPU *p2 = new CPU();
  Score *s = new Score();
  Game *g = new Game();

  g->compareMoves(p1, p2, s);

  return 0;
}

File: RockPaperScissors/Python/RockPaperScissors.py
# Rock Paper Scissors game written in Python
from HumanClass import *
from ScoreClass import Score

# Function that runs the game
def game(score):
    p1 = Human()
    p2 = CPU()

    print(p1.returnMove().upper() + " VERSUS " + p2.returnMove().upper())

    # Determines the winner by checking each possible human move
    # versus each possible computer move
    if p1.returnMove() == "rock":
        if p2.returnMove() == "paper":
            score.addCPU()
        elif p2.returnMove() == "scissors":
            score.addHuman()
        else:
            print("It's a tie!")
    elif p1.returnMove() == "paper":
        if p2.returnMove() == "scissors":
            score.addCPU()
        elif p2.returnMove() == "rock":
            score.addHuman()
        else:
            print("It's a tie!")
    elif p1.returnMove() == "scissors":
        if p2.returnMove() == "rock":
            score.addCPU()
        elif p2.returnMove() == "paper":
            score.addHuman()
        else:
            print("It's a tie!")

def main(score):
    # Initialize the players and score
    play = True

    # Continue playing until the player enters "no"
    while play:
        game(score)
        score.displayScore()
        choice = input("Play again?\n")
        if choice.lower() == "no":
            play = False

if __name__ == '__main__':
    score = Score()
    main(score)
